pkg/service/node.go

```diff
@@ -160,8 +162,37 @@ func (n *NodeService) NodeGetVolumeStats(context.Context, *csi.NodeGetVolumeStat
 	panic("implement me")
 }
 
-func (n *NodeService) NodeExpandVolume(context.Context, *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	panic("implement me")
+func (n *NodeService) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
+	klog.Infof("Resizing filesystem mounted on %s", req.VolumePath)
+	fsType := req.VolumeCapability.GetMount().FsType
+	klog.Infof("fstype: %s", fsType)
+
+	var resizeCmd string
+	if strings.HasPrefix(fsType, "ext") {
+		resizeCmd = "resize2fs"
+	}
+	if strings.HasPrefix(fsType, "xfs") {
+		resizeCmd = "xfs_growfs"
+	}
+
+	if resizeCmd == "" {
+		return nil, status.Error(codes.InvalidArgument, "fsType is neither xfs or ext[234]")
+	}
+
+	device, err := getDeviceByMountPoint(req.VolumePath)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := exec.Command(resizeCmd, device)
+	err = cmd.Run()
+	exitError, incompleteCmd := err.(*exec.ExitError)
```

Comment:
@janosdebugs can you please take a look at the error handling here and tell me what you think about it?

Reply:
Generally, this would be ok, but I would strongly recommend using `errors.As` instead of a type assertion. That one will work even if the error will be wrapped in future.